packages/@uppy/image-editor/src/Editor.js

```diff
@@ -7,6 +7,11 @@ module.exports = class Editor extends Component {
       this.imgElement,
       this.props.opts.cropperOptions
     )
+    this.imgElement.addEventListener('crop', (ev) => {
+      this.rotateRange.value = ev.detail.rotate
+      this.rotateRange.parentNode
+        .setAttribute('aria-label', `${ev.detail.rotate}º`)
```

Comment:
Would these be feasible to do using react state updates, vs updating the dom directly?

Reply:
Also could we attach the event listener in the render call?

Reply:
Bear in mind that I don't have much experience with Uppy so this might not be helpful feedback

Reply:
> Would these be feasible to do using react state updates, vs updating the dom directly?

I suppose it would be possible, but probably less efficient? Currently this component is stateless, so I thought I should keep it this way.

> could we attach the event listener in the render call?

On the render call, the element hasn't been added to the DOM yet – in fact `this.imgElement` is still `undefined` at this point. A "more correct" time to do it would be when the `ref` is assigned, so it can survive re-rendering (but as long as the component is stateless, I suppose that's not really an issue).

Reply:
I think the event listener position makes sense as is, it's listening for a custom event that's only available once the `Cropper` instance is created, so  it seems good to add the listener at that time.

I think the aria-label ideally _should_ be a state value. In the current approach, if the ImageEditor component is rerendered by its parent for some reason, it could reset the aria-label to the default text.

Reply:
> In the current approach, if the ImageEditor component is rerendered by its parent for some reason, it could reset the value to the default mid-interaction.

Maybe I'm reading this wrong, but if the component was re-rendered in a way that forced the DOM elements to be replaced, the `Cropper` instance would also be lost (or rather bound to the former image element, removed from the DOM), which would reset the image to its initial appearance anyway.
That being said, I agree using state would make a cleaner implementation, so working on it.

Reply:
A rerender wouldn't cause component state or properties to be lost. (unless the parent moves the component around or provides a `key={}` in which case it's explicitly asking for state to be blown away)

But editing the properties in the DOM directly causes the DOM and virtual dom to go out of sync, and depending on how preact implements its diffing algorithm, it could decide to replace the value in the DOM by the value in the virtual DOM (which would be 0 or null or whatever the default is.)